tern/analyze/common.py

```diff
@@ -684,3 +684,31 @@ def get_git_toplevel(path):
     except (subprocess.CalledProcessError, FileNotFoundError):
         logger.debug("Cannot find git repo toplevel, path is %s", path)
     return path_to_toplevel
+
+
+def find_installed_pkg_from_shell_script(shell_script):
+    '''Given a shell script, first split it into statements, and then
+    filter the install commands to get the installed software.'''
+    statements = general.split_shell_script(shell_script)
+    command_list = []
+    # traverse the statements, pick out the loop and commands.
+    for stat in statements:
+        if 'command' in stat:
+            command_list.append(Command(stat['command']))
+        elif 'loop' in stat:
+            loop_stat = stat['loop']['loop_statements']
+            for st in loop_stat:
+                if 'command' in st:
+                    command_list.append(Command(st['command']))
+    # filter commands
+    report = ''
```

Comment:
I just paste the codes of `filter_install_commands` in `tern/analyze/common.py`. We can make modifications on the report here to give more specifice reports on the installed software.

Reply:
This looks good!

Reply:
Actually the report is good enough to tell us the ignored commands and unrecognized commands. For example:
```
Ignored Commands:apt-get update

Unrecognized Commands:rm -rf /var/lib/apt/lists/*
```
Maybe there is no need to modify the report here.